lib/utils/datahub.js

```diff
@@ -274,6 +336,23 @@ class DataHub extends EventEmitter {
   }
 }
 
+function generateDpForFlow(flowSpec, dpJson = {}) {
+  let descriptor = lodash.cloneDeep(dpJson)
+  let resources = descriptor.resources || []
+  if (resources.length > 0) {
+    return descriptor
+  }
+  if (lodash.isEmpty(descriptor)) {
+    descriptor.name = flowSpec.meta.dataset
+  }
+  const resourceMapping = flowSpec.inputs[0].parameters['resource-mapping']
```

Comment:
what if `resource-mapping` property is empty? E.g., like here http://api.datahub.io/source/90998f7f90e086bd5fc7c9075dfda43b/historical-income-households-us/successful

Reply:
@anuveyatsu how would you define location (URL) of source data. It should be either in 
* datapackage.json - in this case that line is not executed at all 
* or in resource mapping - in this case it will not be empty

Reply:
@zelima as you can see in the example I've provided, it is in the descriptor in `flow.yaml`. May be it's wrong?

Reply:
@anuveyatsu well this change is about avoiding that. You don't have to define descriptor in `flow.yaml` any more. but we can consider that case as well:
```
if flow.yaml has descriptor property:
  return descriptor 
```

Will add